apps/docs/src/guide/transactions/transaction-request.md

```diff
@@ -0,0 +1,51 @@
+
+<<< @/../../docs-snippets/src/guide/transactions/transaction-request.test.ts#transaction-request-1{ts:line-numbers}
+
+A `CreateTransactionRequest` is used for create transactions, which are transactions that create a new contract on the blockchain.
+
+<<< @/../../docs-snippets/src/guide/transactions/transaction-request.test.ts#transaction-request-2{ts:line-numbers}
+
+> **Note**: We recommend you use the `ContractFactory` for contract deployment as this will shape the create transaction request for you. Information on this can be found in the [contract deployment guide](../contracts/deploying-contracts.md#4-deploying-the-contract).
+
+## Modifying a Transaction Request
+
+Once you have instantiated a transaction request, you can modify it by setting the transaction parameters and policies. This can either be done manually by directly altering the transaction request object, or through helper methods that are available on the above classes.
+
+### Adding Resources to a Transaction Request
+
+Resources populate the inputs and outputs of a transaction request. This can take the form of coins, messages or contracts. The SDK provides a range of methods for dealing with resources. Below will detail how coins and messages can be added to a transaction request.
+
+<<< @/../../docs-snippets/src/guide/transactions/transaction-request.test.ts#transaction-request-3{ts:line-numbers}
+
+### Adding a Contract to a Transaction Request
+
+Scripts can perform multiple actions on chain, therefore you may want to chain contract calls. For this you will need to add a contract to the transaction request. This can be done like so:
+
+<<< @/../../docs-snippets/src/guide/transactions/transaction-request.test.ts#transaction-request-4{ts:line-numbers}
+
+### Adding a Predicate to a Transaction Request
+
+Predicates are used to define the conditions under which a transaction can be executed. Therefore you may want to add a predicate to a transaction request to unlock funds that are utilized by a script. This can be added like so:
+
+<<< @/../../docs-snippets/src/guide/transactions/transaction-request.test.ts#transaction-request-5{ts:line-numbers}
+
+### Adding a Witness and Signing a Transaction Request
+
+The SDK provides a way of either modifying the witnesses for a transaction request directly, or by passing accounts. This will then signing the transaction request with the account's private key. Below will detail how to add a witness to a transaction request:
+
+<<< @/../../docs-snippets/src/guide/transactions/transaction-request.test.ts#transaction-request-6{ts:line-numbers}
+
+### Getting the Transaction ID for a Transaction Request
+
+The transaction ID is a SHA-256 hash of the entire transaction request. This can be useful for tracking the transaction on chain. To get the transaction ID, you can use the following method:
+
+<<< @/../../docs-snippets/src/guide/transactions/transaction-request.test.ts#transaction-request-7{ts:line-numbers}
```

Comment:
We should say here that if we change the TX request it will result in changing the TX ID.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1961/commits/5fcb542f03019c177c4234ea07a05c343dddd034